binding/form_mapping.go

```diff
@@ -182,6 +182,34 @@ func trySetCustom(val string, value reflect.Value) (isSet bool, err error) {
 	return false, nil
 }
 
+func trySplit(vs []string, field reflect.StructField) (newVs []string, err error) {
+	cfTag := field.Tag.Get("collection_format")
+	if cfTag == "" || cfTag == "multi" {
+		return vs, nil
+	}
+
+	var sep string
+	switch cfTag {
+	case "csv":
+		sep = ","
+	case "ssv":
+		sep = " "
+	case "tsv":
+		sep = "\t"
+	case "pipes":
+		sep = "|"
+	default:
+		return vs, fmt.Errorf("%s is not supported in the collection_format. (csv, ssv, pipes)", cfTag)
+	}
+
+	newVs = make([]string, 0)
```

Comment:
```go
	totalLength := 0
	for _, v := range vs {
		totalLength += strings.Count(v, sep) + 1
	}
	newVs := make([]string, 0, totalLength)
	for _, v := range vs {
		newVs = append(newVs, strings.Split(v, sep)...)
	}
```

Benchmark

```sh
goos: darwin
goarch: arm64
BenchmarkSplitStringsOriginal-10         1258774               924.5 ns/op
BenchmarkSplitStringsOptimized-10        1716301               697.3 ns/op
PASS
```

Reply:
I replaced it with the code you mentioned.

Reply:
@slowhigh Thanks